pywbem_mock/_namespaceprovider.py

```diff
@@ -0,0 +1,290 @@
+        # TODO: Somehow we need to get this out of the way.
+        new_namespace = new_namespace.strip('/')
+
+        # Write it back to the instance in case it was changed
+        new_instance['Name'] = new_namespace
+
+        # These values must match those in
+        # tests/unittest/utils/wbemserver_mock.py.
+        try:
+            if new_instance['CreationClassName'] != new_instance.classname \
+                or new_instance['ObjectManagerName'] != OBJECTMANAGERNAME \
+                or new_instance['ObjectManagerCreationClassName'] != \
+                'CIM_ObjectManager' \
+                or new_instance['SystemName'] != SYSTEMNAME \
+                or new_instance['SystemCreationClassName'] != \
+                    SYSTEMCREATIONCLASSNAME:
+
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Namespace creation via CreateInstance: "
+                            "Invalid property for CIM_Instance {0!A}",
+                            new_instance))
+
+        except KeyError as ke:
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                _format("Namespace creation via CreateInstance: "
+                        "Missing property for CIM_Instance {0!A}. Exception "
+                        "{1!A}", new_instance, ke))
+
+        if new_namespace not in self.cimrepository.namespaces:
+            self.cimrepository.add_namespace(new_namespace)
+
+        # Pass NewInstance to the default CreateInstance
+        # Default CreateInstance creates path, validates all key properties
+        # in instance and inserts instance into the repository.
+        return super(CIMNamespaceProvider, self).CreateInstance(namespace,
+                                                                new_instance)
+
+    def ModifyInstance(self, ModifiedInstance,
+                       IncludeQualifiers=None, PropertyList=None):
+        """
+        Modification of CIM_Namespace instance not allowed
+        """
+        raise CIMError(
+            CIM_ERR_NOT_SUPPORTED,
+            _format("Modification of CIM_Namespace {0!A} not implemented. ",
+                    ModifiedInstance.classname))
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Delete the namespace defined by the Name property in InstanceName
+        and remove the instance defined by InstanceName
+        """
+        remove_namespace = InstanceName.keybindings['Name']
+
+        # Reflect the namespace deletion in the CIM repository
+        # This will check the namespace for being empty.
+        self.remove_namespace(remove_namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store = self.get_instance_store(InstanceName.namespace)
+        instance_store.delete(InstanceName)
+
+    @staticmethod
+    def install_provider(conn, interop_namespace, pragma_file,
+                         verbose=None):
```

Comment:
I don't like the way this is done.
It does not lend itself to be a general concept for other providers, plus the difficulty is to understand that it performs the register_provider() call as well.

I think we should discuss what other options could be.
**DISCUSSION**

Reply:
It would be logical to make this part of provider registration in place of this method calling provider registration.

DISCUSSION

Reply:
We agreed to make the idea of a setup for the provider a function of the provider_registration. Thus, the only effort to register a provider is a) create the provider object, b) execute provider_registration

TODO

Reply:
I moved this one to a new PR so separate it from the mass we already have here.  Hopefully that PR can be submitted today.